x/node_event_api_client.go

```diff
@@ -97,6 +100,48 @@ type NodeEventAPIClient struct {
 	Errors chan error
 }
 
+// NodeEventAPIClientSubscription holds any error that happened when trying to subscribe to an events.
+// It also allows to close the subscription to cleanly unsubscribe from the node.
+type NodeEventAPIClientSubscription struct {
+	mqttClient mqtt.Client
+	topic      string
+	error      error
+}
+
+func newSubscription(client mqtt.Client, topic string) *NodeEventAPIClientSubscription {
+	return &NodeEventAPIClientSubscription{
+		mqttClient: client,
+		topic:      topic,
+	}
+}
+
+func newSubscriptionWithError(err error) *NodeEventAPIClientSubscription {
+	return &NodeEventAPIClientSubscription{
+		error: err,
+	}
+}
+
+// Error holds any error that happened when trying to subscribe.
+func (s *NodeEventAPIClientSubscription) Error() error {
+	return s.error
+}
+
+// Close allows to close the subscription to cleanly unsubscribe from the node.
+// It returns ErrNodeEventAPIClientSubscriptionAlreadyClosed if called multiple times.
+func (s *NodeEventAPIClientSubscription) Close() error {
```

Comment:
Could it maybe be necessary to add some mutexes? In case clients use it in a multithreaded way?

Reply:
I thought about it, but nothing in the lib is thread-safe per se, so I skipped it for this.
I checked and I could remove this error and just let the user call Close multiple times. The underlying lib won't complain if you unsubscribe more than once

Reply:
Yeah, sounds like a good alternative.